Servers/UI/OJS.Servers.Ui/ClientApp/src/pages/contest/ContestDetailsPage.tsx

```diff
@@ -152,45 +176,53 @@ const ContestDetailsPage = () => {
                         </>
                     )
                     : null}
-                <LinkButton
-                  id="button-card-compete"
-                  internalClassName={competableOnlyForAdminAndLecturers
-                      ? styles.adminAccessibleButton
-                      : ''}
-                  to={getParticipateInContestUrl({
-                      id: contestIdToNumber,
-                      participationType: ContestParticipationType.Compete,
-                  })}
-                  text="Compete"
-                  state={
-                      canAccessCompeteButton
-                          ? ButtonState.enabled
-                          : ButtonState.disabled
-                }
-                />
-                <LinkButton
-                  id="button-card-practice"
-                  internalClassName={praticableOnlyForAdminOrLecturers
-                      ? styles.adminAccessibleButton
-                      : ''}
-                  to={getParticipateInContestUrl({
-                      id: contestIdToNumber,
-                      participationType: ContestParticipationType.Practice,
-                  })}
-                  text="Practice"
-                  type={LinkButtonType.secondary}
-                  state={
-                      canAccessPracticeButton
-                          ? ButtonState.enabled
-                          : ButtonState.disabled
-                }
-                />
+                {canAccessCompeteButton
+                    ? (
+                        <LinkButton
+                          id="button-card-compete"
+                          internalClassName={competableOnlyForAdminAndLecturers
+                              ? styles.adminAccessibleButton
+                              : ''}
+                          to={getParticipateInContestUrl({
+                              id: contestIdToNumber,
+                              participationType: ContestParticipationType.Compete,
+                          })}
+                          text="Compete"
+                          state={
+                        canAccessCompeteButton
```

Comment:
Buttons dont get rendered if the respective participation is not available. What is the point of this state for the button?

Reply:
We should check here if the contest is official and not if it can be accessed.